src/components/lib/util/heap.h

```diff
@@ -1,34 +1,234 @@
+			}                                                              \
+			assert(n < h->e);                                              \
+			if (CMP_FN(h->data[c], h->data[n])) break;                     \
+			NAME##_swap_entries(h->data, n, c);                            \
+			c = n;                                                         \
+		}                                                                  \
+		return c;                                                          \
+	}                                                                      \
+                                                                           \
+	static inline int NAME##_swap_up(heap_t *h, int c)                     \
+	{                                                                      \
+		assert(c <= h->e);                                                 \
+		assert(c > 0);                                                     \
+                                                                           \
+		while (c > 1) {                                                    \
+			int p = c / 2;                                                 \
+			assert(p != 0);                                                \
+			if (CMP_FN(h->data[p], h->data[c])) break;                     \
+			NAME##_swap_entries(h->data, p, c);                            \
+			c = p;                                                         \
+		}                                                                  \
+		assert(c != 0);                                                    \
+		return c;                                                          \
+	}                                                                      \
+                                                                           \
+	static inline int NAME##_heapify(heap_t *h, int c)                     \
+	{                                                                      \
+		c = NAME##_swap_up(h, c);                                          \
+		return NAME##_swap_down(h, c);                                     \
+	}                                                                      \
+                                                                           \
+	static inline int NAME##_heap_verify(heap_t *h, int c)                 \
+	{                                                                      \
+		int left, right;                                                   \
+                                                                           \
+		left  = c * 2;                                                     \
+		right = c * 2 + 1;                                                 \
+		if (left < h->e) {                                                 \
+			assert(((hentry_t *)h->data[left])->index == left);            \
+			if (!CMP_FN(h->data[c], h->data[left]) || NAME##_heap_verify(h, left)) {        \
+				printc("Left data %d @ %d < %d @ %d\n", ((hentry_t *)h->data[c])->value, c, \
```

Comment:
Do you have prints in the code? We shouldn't commit code with debugging prints, if that's what this is (and the one below).

Reply:
This is the function that the unit test (heap_user.c) uses to verify, it only prints and asserts in case of failure. 
Would it be better to move this under heap_user.c function and hard-code the CMP_FN?

Reply:
Ooops, that's my bad. I forgot the context of these prints.

I'm sorry, they are fine! Ignore my comment.